test/com/gfredericks/test/chuck/exception_handling_test.cljc

```diff
@@ -0,0 +1,63 @@
+(ns com.gfredericks.test.chuck.exception-handling-test
+  (:require #?(:clj  [clojure.test :refer :all]
+               :cljs [cljs.test :as test :refer [test-var test-vars *current-env*]
+                      :refer-macros [is testing deftest]])
+            [clojure.test.check.generators :as gen]
+            [com.gfredericks.test.chuck.clojure-test #?(:clj :refer :cljs :refer-macros) [checking]]))
+
+(deftest this-test-should-crash-and-be-caught
+  (checking "you can divide four by numbers" 100 [i gen/pos-int]
+    ;; going for uncaught-error-not-in-assertion here
+    (let [n #?(:clj  (/ 4 i)
+               :cljs (throw (js/Error. "Oops!")))]
+      (is n))))
+
+(defn capture-test-var [v]
+  (with-out-str (test-var v)))
+
+(deftest exception-detection-test
+  (let [test-results
+        #?(:clj
+           (binding [; need to keep the failure of this-test-should-crash-and-be-caught
+                     ; from affecting the clojure.test.check test run
+                     *report-counters* (ref *initial-report-counters*)]
+             (capture-test-var #'this-test-should-crash-and-be-caught)
+             @*report-counters*)
+
+           :cljs
+           (binding [*current-env* (test/empty-env)]
+             (capture-test-var #'this-test-should-crash-and-be-caught)
+             (:report-counters *current-env*)))]
```

Comment:
the logic represented by this `#?(...)` in duplicated in the next test, and is hard to read in this context. Could it be extracted into a function whose name describes what it's doing? (which I assume is running the test and returning the results)